src/fidesops/api/v1/endpoints/user_endpoints.py

```diff
@@ -17,16 +18,28 @@
 from fidesops.util.oauth_util import verify_oauth_client
 from sqlalchemy.orm import Session
 
-from fidesops.api.v1.scope_registry import (
-    USER_CREATE,
-    USER_DELETE,
-    SCOPE_REGISTRY,
-)
+from fidesops.api.v1.scope_registry import USER_CREATE, USER_DELETE, SCOPE_REGISTRY
 
 logger = logging.getLogger(__name__)
 router = APIRouter(tags=["Users"], prefix=V1_URL_PREFIX)
 
 
+def perform_login(db: Session, user: FidesopsUser) -> ClientDetail:
+    """Performs a login by updating the FidesopsUser instance and
+    creating and returning an associated ClientDetail."""
+    user.last_login_at = datetime.utcnow()
+
+    client: ClientDetail = user.client
+    if not client:
+        logger.info("Creating client for login")
+        client, _ = ClientDetail.create_client_and_secret(
+            db, SCOPE_REGISTRY, user_id=user.id
+        )
+
+    user.save(db)
```

Comment:
minor, only leaving this comment since we need the migration update too - I'd group "like" functionality together, so saving the user above on L31 instead of on L39.